packages/app/src/client/views/integrations/all/Install.tsx

```diff
@@ -50,37 +44,56 @@ export const InstallIntegration = () => {
   const dispatch = useDispatch();
   const history = useHistory();
   const tenant = useSelectedTenantWithFallback();
+  const { registerNotification } = useSimpleNotification();
 
   const integration = integrationDefRecords[kind];
   if (!integration) return <NotFound />;
 
-  const onCreate = (data: NewIntegration, options?: NewIntegrationOptions) => {
-    graphqlClient
-      .InsertIntegration({
+  const onCreate = async <IntegrationData,>(
+    data: NewIntegration<IntegrationData>,
+    options?: NewIntegrationOptions
+  ) => {
+    let response;
+    try {
+      response = await graphqlClient.InsertIntegration({
         name: data.name,
         kind: kind,
         data: data.data || {},
         tenant_id: tenant.id
-      })
-      .then(response => {
-        const integration = response?.data?.insert_integration_one;
-        if (integration) {
-          dispatch(addIntegration({ integration }));
-
-          if (options?.createGrafanaFolder) {
-            createFolder({ integration, tenant }).then(() =>
-              dispatch(loadGrafanaStateForIntegration({ id: integration.id }))
-            );
-          }
+      });
+    } catch (error) {
+      registerNotification({
+        state: "error" as const,
+        title: "Could not install integration",
+        information: error.response.errors?.[0].message ?? error.message
```

Comment:
is it possible for `error.response.errors` to be an empty array?

Reply:
At least not according to the [graphql-spec](https://github.com/graphql/graphql-spec/blob/main/spec/Section%207%20--%20Response.md#errors)
> The errors entry in the response is a non-empty list of errors, where each error is a map.

Reply:
Meaning that if `errors` is not `undefined`, it is a non-empty array.

Checking for the `errors` key is a real important concept when getting GraphQL responses -- I also learned about that only recently, and added corresponding code commentary here:

https://github.com/opstrace/opstrace/blob/5ade2201e34d35bf60894e42ae4f7d67968a1490/test/test-remote/test_ui_api.ts#L451

Reply:
@jgehrcke @MatApple 
Looking at the code, I think it might be enough to just read `error.message` - because the library we're using ([graphql-request](https://github.com/prisma-labs/graphql-request)) does a decent job covering [handling the errors](https://github.com/prisma-labs/graphql-request/blob/master/src/types.ts#L25-L55) and covering most cases, [including the one @jgehrcke  mentioned](https://github.com/prisma-labs/graphql-request/blob/master/src/types.ts#L49-L53).

The question is, if we're happy with [their error message format](https://github.com/prisma-labs/graphql-request/blob/master/src/types.ts#L30-L33), which seems to be more geared toward logging than presenting to user.

Reply:
BTW - yes, GraphQL will return 200, even if there are field errors. Our client library will throw regardless.